grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/CorenessQueryImpl.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.query.analytics;
+
+import ai.grakn.GraknTx;
+import ai.grakn.concept.Label;
+import ai.grakn.concept.LabelId;
+import ai.grakn.concept.SchemaConcept;
+import ai.grakn.concept.Type;
+import ai.grakn.exception.GraqlQueryException;
+import ai.grakn.graql.analytics.CorenessQuery;
+import ai.grakn.graql.internal.analytics.CorenessVertexProgram;
+import ai.grakn.graql.internal.analytics.DegreeDistributionMapReduce;
+import ai.grakn.graql.internal.analytics.NoResultException;
+import ai.grakn.graql.internal.util.StringConverter;
+import com.google.common.collect.Sets;
+import org.apache.tinkerpop.gremlin.process.computer.ComputerResult;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.joining;
+
+class CorenessQueryImpl extends AbstractComputeQuery<Map<Integer, Set<String>>> implements CorenessQuery {
+
+    private int k = 2;
+    private boolean ofTypeLabelsSet = false;
+    private Set<Label> ofLabels = new HashSet<>();
+
+    CorenessQueryImpl(Optional<GraknTx> graph) {
+        this.tx = graph;
+    }
+
+    @Override
+    public Map<Integer, Set<String>> execute() {
+        LOGGER.info("Coreness query is started");
+        long startTime = System.currentTimeMillis();
+
+        if (k < 2) throw GraqlQueryException.kValueSmallerThanTwo();
+
+        includeAttribute = true; //TODO: REMOVE THIS LINE
```

Comment:
Can we remove it now?

Reply:
Not yet. I will do it in another pr, so we can remove it from all analytics queries